src/Engine/ProtoCore/AssociativeGraph.cs

```diff
@@ -1259,6 +1262,28 @@ public List<GraphNode> GraphList
         }
 
         /// <summary>
+        /// Mark graphnodes in scope as dirty
+        /// </summary>
+        /// <param name="block"></param>
+        /// <param name="classIndex"></param>
+        /// <param name="procIndex"></param>
+        public void MarkGraphNodesDirty(int block, int classIndex, int procIndex)
+        {
+            List<GraphNode> gnodeList = GetGraphNodesAtScope(classIndex, procIndex);
+            if (gnodeList != null)
+            {
+                foreach (GraphNode gnode in gnodeList)
+                {
+                    if (gnode.languageBlockId == block)
+                    {
+                        gnode.isDirty = true;
+                    }
+                }
+            }
+        }
+
```

Comment:
This overloaded `MarkGraphNodesDirty()` implementation is quite different from other versions which checks
  - If a node is a binary expression
  - Check `OriginalAstID`
  - Set `GraphNode.IsActive`.

Are those extra checking and settings needed in this overloaded function as well?